apps/src/code-studio/components/StarWarsGridCellEditor.jsx

```diff
@@ -18,47 +19,41 @@ const options = {
   covered_crates_3b: 0x350000,
 };
 
-export default class StarWarsGridCellEditor extends React.Component {
-  static propTypes = {
-    cell: React.PropTypes.object.isRequired,
-    row: React.PropTypes.number.isRequired,
-    col: React.PropTypes.number.isRequired,
-    onUpdate: React.PropTypes.func.isRequired,
-  }
-
-  constructor(props) {
-    super(props);
-    this.handleChange = this.handleChange.bind(this);
-  }
-
+export default class StarWarsGridCellEditor extends CellEditor {
+  /**
+   * @override
+   */
   handleChange() {
-    const zoom = this.refs.zoom && this.refs.zoom.checked;
+    const zoom = this.zoom && this.zoom.checked;
     this.props.onUpdate({
-      tileType: zoom << WallTypeShift | this.refs.type.value
+      tileType: zoom << WallTypeShift | this.type.value
     });
   }
 
+  /**
+   * @override
+   */
   render() {
     const type = ~WallTypeMask & this.props.cell.tileType_;
     const zoom = WallTypeMask & this.props.cell.tileType_;
 
     return (
       <form className="span4 offset1">
-        <img src={tileGuide}/>
+        <img src={tileGuide} />
         <header>
           <strong>Editing Cell ({this.props.row}, {this.props.col})</strong>
         </header>
 
         <label htmlFor="tileType">Tile Type (required):</label>
-        <select ref="type" name="tileType" value={type} onChange={this.handleChange}>
+        <select ref={c => { this.type = c; }} name="tileType" value={type} onChange={this.handleChange}>
```

Comment:
It's the new preferred way of saving references; the old string-based refs are considered legacy: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md#prevent-using-string-references-reactno-string-refs

https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs